messaging/tck/src/main/java/org/eclipse/microprofile/reactive/messaging/tck/client/TckDeployListener.java

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package org.eclipse.microprofile.reactive.messaging.tck.client;
+
+import org.eclipse.microprofile.reactive.messaging.tck.container.Topics;
+import org.eclipse.microprofile.reactive.messaging.tck.client.event.DeployTopics;
+import org.eclipse.microprofile.reactive.messaging.tck.client.event.TopicEvent;
+import org.eclipse.microprofile.reactive.messaging.tck.client.event.UnDeployTopics;
+import org.jboss.arquillian.container.spi.client.container.DeploymentException;
+import org.jboss.arquillian.container.spi.event.container.AfterUnDeploy;
+import org.jboss.arquillian.container.spi.event.container.BeforeDeploy;
+import org.jboss.arquillian.container.test.impl.client.deployment.event.GenerateDeployment;
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.InstanceProducer;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.arquillian.test.spi.annotation.ClassScoped;
+
+import java.util.Arrays;
+
+public class TckDeployListener {
+
+    @Inject
+    @ClassScoped
+    private InstanceProducer<TopicDeploymentScenario> topicDeploymentScenario;
+
+    @Inject
+    private Event<TopicEvent> topicEvent;
+
+    public void onGenerateDeployment(@Observes GenerateDeployment event) {
```

Comment:
Not sure if this is a public event I should be hooking into, but since it's the event that goes and locates the `@Deployment` methods, I thought it would also be the appropriate place to locate the `@Topic` declarations.